src/adapters/philips_hue/hub_api.rs

```diff
@@ -88,13 +96,63 @@ impl HubApi {
         !settings.contains("unauthorized user")
     }
 
-    pub fn try_pairing(&self) -> bool {
-        // [{"success":{"username":"foxboxb-001788fffe25681a"}}]
-        // [{"error":{"type":101,"address":"/","description":"link button not pressed"}}]
+    pub fn try_pairing(&self) -> Result<Option<String>, ()> {
+        #[derive(Deserialize, Debug)]
+        struct PairingResponse {
+            success: Option<SuccessResponse>,
+            error: Option<ErrorResponse>
+        }
+        #[derive(Deserialize, Debug)]
+        struct SuccessResponse {
+            username: String
+        }
+        #[derive(Deserialize, Debug)]
+        struct ErrorResponse {
+            #[serde(rename="type")]
+            error_type: u32,
+            address: String,
+            description: String
+        }
         let url = "api";
-        let req = json!({ username: self.token, devicetype: "foxbox_hub"});
-        let response = self.post_unauth(&url, &req).unwrap_or("".to_owned()); // TODO: no unwrap
-        response.contains("success")
+        let req = json!({ devicetype: "foxbox_hub"});
+        let response = self.post_unauth(&url, &req).unwrap_or("[]".to_owned());
+        let mut response: Vec<PairingResponse> =
+            structs::parse_json(&response).unwrap_or(Vec::new());
+        if response.len() != 1 {
+            error!("Pairing request to Philips Hue bridge {} yielded unexpected response",
+                self.id);
+            return Err(())
+        }
+        let response = match response.pop() {
+            Some(response) => { response },
+            None => { return Err(()) }
+        };
+        match response.success {
+            Some(success) => {
+                Ok(Some(success.username))
+            },
```

Comment:
We can get slightly more readable code using `if let`:
if let Some(success) = response.success {
    Ok(Some(success.username))
} else {
    ...
}